File: LMS/LMS.Application/DTOs/HR/Employees/EmployeeDetailsDto.cs
using LMS.Application.DTOs.Financial;
using LMS.Application.DTOs.HR.Departments;
using LMS.Domain.HR.Models;

namespace LMS.Application.DTOs.HR.Employees
{
    public class EmployeeDetailsDto
    {
        public Guid EmployeeId { get; set; }
        public string FullName { get; set; } = string.Empty;
        public string PhoneNumber { get; set; } = string.Empty;
        public DateTime HireDate { get; set; }
        public string Email { get; set; } = string.Empty;
        public decimal BaseSalary { get; set; }
        public string? CurrentDepartmentName { get; set; } = string.Empty;

        public ICollection<RevenueOverviewDto> EmployeeFinanical { get; set; } = [];
        public ICollection<DepartmentHistoryDto> DepartmentsHistory { get; set; } = [];
        public ICollection<AttendanceLookup> Attendances { get; set; } = [];
        public ICollection<IncentivesLookupDto>  Incentives { get; set; } = [];
        public ICollection<PenaltyLookupDto>  Penalties { get; set; } = [];
        public ICollection<LeaveOverviewDto>  Leaves { get; set; } = [];
        public ICollection<SalariesOverviewDto>  Salaries { get; set; } = [];
    }
}

File: LMS/LMS.UI/huda-center/src/app/shared/models/hr/employee/employee-details.dto.ts
import { AttendanceLookup } from "../../hr/attendance-view.dto";
import { RevenueOverviewDto } from "../../finacials/financial-revenue.dto";
import { IncentivesLookupDto } from "../../hr/inecntive-view.dto";
import { LeaveOverviewDto } from "../../hr/leave-view.dto";
import { PenaltyLookupDto } from "../../hr/penalty-view.dt";
import { SalariesOverviewDto } from "../../hr/salary-view.dto";
import { DepartmentHistoryDto } from "../departments/department-history.dto";

export interface EmployeeDetailsDto {
    employeeId: string;
    fullName: string;
    phoneNumber: string;
    hireDate: Date;
    email: string;
    baseSalary: number;
    currentDepartmentName: string;
    departmentsHistory: DepartmentHistoryDto[];
    employeeFinanical: RevenueOverviewDto[];
    attendances: AttendanceLookup[];
    incentives: IncentivesLookupDto[];
    penalties: PenaltyLookupDto[];
    leaves: LeaveOverviewDto[];
    salaries: SalariesOverviewDto[];
}